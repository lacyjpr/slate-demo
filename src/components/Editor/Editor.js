import React, { Component } from 'react';

class Editor extends Component {
  bold = () => {
    document.execCommand('bold', false, null);
  };

  render() {
    return (
      <div>
        <button onClick={this.bold}>bold</button>
        <div contentEditable="true">This text can be edited by the user.</div>
      </div>
    );
  }
}

export default Editor;
