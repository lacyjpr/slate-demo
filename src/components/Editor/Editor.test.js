import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Editor from './Editor';

describe('<Editor />', () => {
  it('renders without crashing', () => {
    shallow(<Editor />);
  });
  it('renders correctly', () => {
    const tree = renderer.create(<Editor />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
