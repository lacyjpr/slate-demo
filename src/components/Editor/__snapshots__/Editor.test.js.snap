// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Editor /> renders correctly 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    bold
  </button>
  <button
    onClick={[Function]}
  >
    italic
  </button>
  <div
    contentEditable="true"
  >
    This text can be edited by the user.
  </div>
</div>
`;
